client/my-sites/plugins/with-plugin-redirect.js

```diff
@@ -0,0 +1,150 @@
+import { transferStates } from 'calypso/state/automated-transfer/constants';
+import { successNotice, removeNotice } from 'calypso/state/notices/actions';
+
+// Plugins list that should perfomr a redirect
+// once it's installed.
+const redirectingPluginsList = {
+	woocommerce: getSiteWoocommerceWizardUrl,
+};
+
+/**
+ * Return the handler function to get the plugin redirect URL.
+ *
+ * @param {string} slug - Plugin slug.
+ * @returns {Function|boolean} Selector handler if the plugin should redirect. Otherwise, False.
+ */
+function getPluginRedirectUrl( slug ) {
+	if ( Object.keys( redirectingPluginsList ).indexOf( slug ) < 0 ) {
+		return false;
+	}
+
+	return redirectingPluginsList[ slug ];
+}
+
+const withPluginRedirect = ( Component ) => ( props ) => {
+	const { plugin = {} } = props;
+	const pluginSlug = plugin?.slug;
+
+	if ( ! pluginSlug ) {
+		return <Component { ...props } />;
+	}
+
+	const redirectUrlHandler = getPluginRedirectUrl( pluginSlug );
+	if ( ! redirectUrlHandler ) {
+		return <Component { ...props } />;
+	}
+
+	const translate = useTranslate();
+	const dispatch = useDispatch();
+
+	const { redirect, url, hasSiteBeenTransferred } = useSelector(
+		( state ) => {
+			const site = getSelectedSite( state );
+			const isAtomic = isSiteAutomatedTransfer( state, site.ID );
+			const isJetpack = isJetpackSite( state, site.ID );
+			const woocommerceWizardUrl = redirectUrlHandler( state, site.ID );
+			const transferState = getAutomatedTransferStatus( state, site.ID );
+			const hasPluginJustBeenInstalled = isPluginActionCompleted(
+				state,
+				site.ID,
+				pluginSlug,
+				INSTALL_PLUGIN
+			);
+
+			const isSiteConnected = getSiteConnectionStatus( state, site.ID );
+			const selectedData = {
+				redirect: false,
+				url: woocommerceWizardUrl,
+				hasSiteBeenTransferred: transferState === transferStates?.COMPLETE,
+			};
+
+			// Jetpack site.
+			if (
+				isJetpack &&
+				pluginSlug === 'woocommerce' &&
+				woocommerceWizardUrl &&
+				hasPluginJustBeenInstalled &&
+				isSiteConnected
+			) {
+				return { ...selectedData, redirect: true };
+			}
+
+			// Atomic Site.
+			if (
+				isAtomic &&
+				pluginSlug === 'woocommerce' &&
+				woocommerceWizardUrl &&
+				selectedData.hasSiteBeenTransferred &&
+				isSiteConnected
+			) {
+				return { ...selectedData, redirect: true };
+			}
+
+			return selectedData;
+		},
+		[ pluginSlug ]
+	);
+
+	const pluginName = plugin?.name;
+
+	useEffect( () => {
+		if ( ! redirect || ! url || ! pluginName ) {
+			return;
+		}
+
+		const pluginInstalledNoticeTime = hasSiteBeenTransferred ? 0 : 3000;
+
+		let timerId = setTimeout( () => {
+			dispatch( removeNotice( 'plugin-notice' ) );
+
+			// re-use same timer ID to redirect the client.
+			timerId = setTimeout( () => {
+				window.location.href = url;
+			}, 5500 );
+
+			dispatch(
+				successNotice(
+					translate( 'Redirecting to the %(pluginName)s onborading page in five seconds.', {
```

Comment:
There's a typo with 'onboarding', but should we mention onboarding at all? Maybe 'Redirecting to setup %{pluginName} in five seconds'?

Reply:
sounds better to me. Also, let's make define it in the plugins list object.